pkg/kubectl/cmd/util/cmdproto/cmdproto.go

```diff
@@ -0,0 +1,88 @@
+package cmdproto
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"time"
+	"unsafe"
+
+	"github.com/golang/protobuf/descriptor"
+	"github.com/golang/protobuf/proto"
+	"github.com/spf13/cobra"
+
+	cmdproto "k8s.io/kubernetes/pkg/kubectl/cmd/util/cmdproto/k8s_io_kubectl_cmd"
+)
+
+func extractFlagDetailFromMessage(i int, msg interface{}) (string, string, []string, string, cmdproto.FlagDetail_EXT, error) {
+	_, md := descriptor.ForMessage(msg.(descriptor.Message))
+	opts := md.Field[i].GetOptions()
+	info, err := proto.GetExtension(opts, cmdproto.E_Info)
+	if err != nil {
+		return "", "", []string{""}, "", cmdproto.FlagDetail_NONE, err
+	}
+	return info.(*cmdproto.FlagDetail).GetName(), info.(*cmdproto.FlagDetail).GetShorthand(), info.(*cmdproto.FlagDetail).GetValue(), info.(*cmdproto.FlagDetail).GetUsage(), info.(*cmdproto.FlagDetail).GetExt(), nil
+}
+
+func FlagsSetup(cmd *cobra.Command, msg interface{}) {
```

Comment:
Would it be possible to restructure this so instead of taking the cmd it returns the cmd, and takes an interface that accepts the `msg` and returns the output + error.